### PR TITLE
Fixing ObjManip HostTransform bugs

### PR DESCRIPTION
**Pull request by Zee2**
## Overview
Fixing #10889. Assigning the HostTransform at runtime now works properly.

Assigning HostTransform in the middle of a manipulation is not currently supported; waiting on Norbert's #11032 PR to go in to fix that behavior.

## Changes
- Makes assigning HostTransform during a manipulation fail with a warning
- Makes assigning HostTransform at runtime properly re-initialize ConstraintManager and re-grabs reference to rigidbody
- Adds unit test to catch the bug.

### CODE REVIEW

**Comment by JPalmerDev** [comment]
Hooray.

Glad to hear this is getting fixed.

File: com.microsoft.mrtk.core/Utilities/Extensions/VectorExtensions.cs [hunk 1]
```diff
@@ -169,7 +169,7 @@ public static bool IsValidVector(this Vector3 vector)
         /// <param name="v2">The second vector.</param>
         /// <param name="distanceTolerance">The maximum distance that will cause this to return true.</param>
         /// <returns>True if the distance between the two vectors is within the tolerance, false otherwise.</returns>
-        public static bool CloseEnough(Vector3 v1, Vector3 v2, float distanceTolerance)
+        public static bool CloseEnoughTo(this Vector3 v1, Vector3 v2, float distanceTolerance = 0.001f)
         {
             return Mathf.Abs(Vector3.Distance(v1, v2)) < distanceTolerance;
```
Comment (keveleigh) [inline com.microsoft.mrtk.core/Utilities/Extensions/VectorExtensions.cs:174]: nit (though outside your actual changes): we shouldn't need to take the absolute value of a distance 🤪

File: com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs [hunk 2]
```diff
@@ -79,7 +79,29 @@ public Transform HostTransform
 
                 return hostTransform;
             }
-            set => hostTransform = value;
+            set
+            {
+                if (interactorsSelecting.Count == 0)
```
Comment (keveleigh) [inline com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs:84]: Should this be `!=` to check for an active manipulation?

Comment (Zee2) [reply com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs:84]: ooooops

File: com.microsoft.mrtk.core/Utilities/Extensions/VectorExtensions.cs (hunk 1 above)
Comment (Zee2) [reply com.microsoft.mrtk.core/Utilities/Extensions/VectorExtensions.cs:174]: Fixed

**Review by keveleigh — APPROVED** [review]

File: com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs (hunk 2 above)
Comment (Zee2) [reply com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs]: Fixed

**Pull request merged**